src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -134,25 +147,39 @@ class WorkflowManagerActor(dataAccess: DataAccess, backend: Backend) extends Act
     }
   }
 
-  private def callStdoutStderr(workflowId: WorkflowId, callFqn: String, index: ExecutionIndex): Future[StdoutStderr] = {
+  private def isCollector(entries: Iterable[ExecutionDatabaseKey])(key: ExecutionDatabaseKey) = {
+    key.index.isEmpty &&
+      (entries exists { e =>
+        (e.fqn == key.fqn) && e.index.isDefined
+      })
+  }
+
+  private def hasLogs(entries: Iterable[ExecutionDatabaseKey])(key: ExecutionDatabaseKey) = {
+    !key.fqn.isScatter && !isCollector(entries)(key)
```

Comment:
Agree w/ your latter point but instead of saying "woops, we can't do it because we did our types stupid" I'd rather either fix FQN now or leave the implicit here so that it gets picked up when we convert all of our types in the future. Otherwise this will just be forgotten in time